src/adapter/search-request-adapter/search-params-adapter.ts

```diff
@@ -1,18 +1,36 @@
-import type { MeiliSearchParams, SearchContext } from '../../types'
+import type {
+  MeiliSearchParams,
+  SearchContext,
+  Filter,
+  PaginationState,
+} from '../../types'
 
 import {
   adaptGeoPointsRules,
   createGeoSearchContext,
 } from './geo-rules-adapter'
 import { adaptFilters } from './filter-adapter'
 
-function setScrollPagination(
-  hitsPerPage: number,
-  page: number,
+function isPaginationRequired(
+  filter: Filter,
   query?: string,
   placeholderSearch?: boolean
+): boolean {
+  if (!placeholderSearch && !query) {
+    if (!filter || filter.length === 0) {
+      return false
+    }
+  }
+  return true
+}
```

Comment:
What happens if `if (!placeholderSearch && !query)` is true, but `if (!filter || filter.length === 0) {` is false?

Reply:
then the pagination is required, both must be true to disable pagination